app/src/main/java/com/example/hackernews/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.hackernews.viewmodel
+
+import android.content.Context
+import androidx.annotation.WorkerThread
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.ViewModel
+import com.example.hackernews.model.entities.User
+import com.example.hackernews.model.repository.UserRepository
+
+typealias ErrorCallBack = (Result) -> Unit
+
+sealed class Result {
```

Comment:
Let's not mix callbacks with Result type. Always use callbacks. If you need to pass data from ViewModel to view please use observable properties (livedata).